lib/matrix_convert_to_zero.rb

```diff
@@ -3,8 +3,57 @@
 # If any number is found to be 0, the method updates all the numbers in the
 # corresponding row as well as the corresponding column to be 0.
 
-# Time complexity: ?
-# Space complexity: ?
+#Approach 1:
+# Time complexity: O(n^3) 3 nested loops
+# Space complexity: O(n^2) n is the number of rows and columns in the matrix being cloned
```

Comment:
This is true if the number of rows and columns are the same (i.e. the input matrix is a "square" matrix).  However, if you allow the number of rows & columns to differ, you probably need to specify `O(nm)` and define `n` as the number of rows and `m` as the number of columns.